src/senselab/utils/data_structures/video.py

```diff
@@ -0,0 +1,69 @@
+"""Data structures relevant for video tasks and pipelines."""
+
+import uuid
+from typing import Dict, Optional
+
+import torch
+from pydantic import BaseModel, Field
+from torchvision.io import read_video
+
+from senselab.utils.constants import SENSELAB_NAMESPACE
+from senselab.utils.data_structures.audio import Audio
+
+
+class Video(BaseModel):
+    """Pydantic model for video and its corresponding metadata.
+
+    Pydantic model for video that holds the necessary attributes, the actual decoded video data
+    and the frame rate, to work with videos in python. Contains metadata information as needed
+    and has a unique identifier for every video.
+
+    Attributes:
+        frames: Represent the video as a Tensor of all of its frames, each of which is an image
+            that we represent through a Tensor of (C, H, W)
+        frame_rate: Also known as the frames per second (fps), defines the time component
+            of a video (often an integer but some use cases of float approximations)
+        audio: the audio associated with the Video (optional)
+        orig_path_or_id: Optional str for the original path or an ID to track file over time
+        metadata: Optional metadata dictionary of information associated with this Video instance
+            (e.g. participant demographics, video settings, location information)
+    """
+
+    frames: torch.Tensor
```

Comment:
Should we validate the shape of this torch.Tensor?

Reply:
Yeah we can, I think it should be 4 dimensions?